spring-batch-integration/src/main/java/org/springframework/batch/integration/partition/RemotePartitioningMasterStepBuilder.java

```diff
@@ -0,0 +1,297 @@
+		Assert.notNull(inputChannel, "inputChannel must not be null");
+		this.inputChannel = inputChannel;
+		return this;
+	}
+
+	/**
+	 * Set the output channel on which requests to workers will be sent.
+	 * The output channel will be set as a default channel on the provided
+	 * {@link MessagingTemplate} trough
+	 * {@link RemotePartitioningMasterStepBuilder#messagingTemplate(MessagingTemplate)}
+	 * (or the one created by this builder if no messaging template is provided).
+	 * @param outputChannel the output channel.
+	 * @return this builder instance for fluent chaining
+	 *
+	 * @see RemotePartitioningMasterStepBuilder#messagingTemplate(MessagingTemplate)
+	 */
+	public RemotePartitioningMasterStepBuilder outputChannel(MessageChannel outputChannel) {
+		Assert.notNull(outputChannel, "outputChannel must not be null");
+		this.outputChannel = outputChannel;
+		return this;
+	}
+
+	/**
+	 * Set the {@link MessagingTemplate} to use to send data to workers.
+	 * <p><strong>The default destination of the messaging template will be
+	 * overridden by the output channel provided through
+	 * {@link RemotePartitioningMasterStepBuilder#outputChannel(MessageChannel)}</strong>.</p>
+	 *
+	 * @param messagingTemplate the messaging template to use
+	 * @return this builder instance for fluent chaining
+	 */
+	public RemotePartitioningMasterStepBuilder messagingTemplate(MessagingTemplate messagingTemplate) {
+		Assert.notNull(messagingTemplate, "messagingTemplate must not be null");
+		this.messagingTemplate = messagingTemplate;
+		return this;
+	}
+
+	/**
+	 * Set the job explorer.
+	 * @param jobExplorer the job explorer to use.
+	 * @return this builder instance for fluent chaining
+	 */
+	public RemotePartitioningMasterStepBuilder jobExplorer(JobExplorer jobExplorer) {
+		Assert.notNull(jobExplorer, "jobExplorer must not be null");
+		this.jobExplorer = jobExplorer;
+		return this;
+	}
+
+	/**
+	 * How often to poll the job repository for the status of the workers. Defaults to 10 seconds.
+	 * @param pollInterval the poll interval value in milliseconds
+	 * @return this builder instance for fluent chaining
+	 */
+	public RemotePartitioningMasterStepBuilder pollInterval(long pollInterval) {
+		Assert.isTrue(pollInterval > 0, "The poll interval must be greater than zero");
+		this.pollInterval = pollInterval;
+		return this;
+	}
+
+	/**
+	 * When using job repository polling, the time limit to wait. Defaults to -1 (no timeout).
+	 * @param timeout the timeout value in milliseconds
+	 * @return this builder instance for fluent chaining
+	 */
+	public RemotePartitioningMasterStepBuilder timeout(long timeout) {
+		this.timeout = timeout;
+		return this;
+	}
+
+	/**
+	 * Set the bean factory.
+	 * @param beanFactory the bean factory to use
+	 * @return this builder instance for fluent chaining
+	 */
+	public RemotePartitioningMasterStepBuilder beanFactory(BeanFactory beanFactory) {
```

Comment:
I think this simply is propagated from the `RemotePartitioningMasterStepBuilderFactory`, so should not be `public`. Isn't it?